packages/terra-application-navigation/src/utils/helpers.js

```diff
@@ -61,6 +61,22 @@ const useAnimatedCount = (countRef, countValue) => {
   }, [countRef, countValue]);
 };
 
+function itemId(appNavId, itemType, itemKey) {
+  return appNavId ? `${appNavId}-${itemType}-${itemKey}` : null;
```

Comment:
Not sure if returning null is the right thing to do here when there is no appNavId. I assume we don't want to enforce ids if the consumer doesn't want them.

Reply:
if the user hasn't entered an app nav id we shouldn't add ids at all.